tx-pool/src/component/pending.rs

```diff
@@ -71,7 +76,15 @@ impl PendingQueue {
                     .expect("exists")
             });
             ancestors.push(id.clone());
-            proposals.extend(ancestors.into_iter().take(limit - proposals.len()));
```

Comment:
In fact, this previous line has a small bug: Consider the below case,

```
limit = 2
proposals = [a]
ancestors = [a, b, c, d]
```

The final `proposals` by previous code is `[a]`, but what we expect is `[a, b]`